fixes.inc

```diff
@@ -14281,6 +14354,133 @@ native BAD_DestroyActor(actorid) = DestroyActor;
 	#define DestroyActor__( FIXES_DestroyActor(
 #endif
 
+/**
+ * <fixes> Various bugs in GetVehicleComponentInSlot function.</fixes>
+ */
+
+#if _FIXES_SAMP && defined _ALS_GetVehicleComponentInSlot
+	#error _ALS_GetVehicleComponentInSlot defined
+#endif
+
+native BAD_GetVehicleComponentInSlot(vehicleid, slot) = GetVehicleComponentInSlot;
+
+#if FIX_GetVehicleComponentInSlot
+
+	stock FIXES_GetVehicleComponentType(componentid)
+	{
+		if(IsComponentFrontBullbar(componentid))
+		{
+			return CARMODTYPE_FRONT_BULLBAR;
+		}
+		else if(IsComponentRearBullbar(componentid))
+		{
+			return CARMODTYPE_REAR_BULLBAR;
+		}
+		return GetVehicleComponentType__(componentid);
+	}
+
+	stock FIXES_AddVehicleComponent(vehicleid, componentid)
+	{
+		switch(GetVehicleComponentType(componentid))
+		{
+			case CARMODTYPE_STEREO:
+			{
+				FIXES_gsStereo[vehicleid] = componentid;
+			}
+			case CARMODTYPE_FRONT_BUMPER:
+			{
+				FIXES_gsBumpers[vehicleid][0] = componentid;
+			}
+			case CARMODTYPE_REAR_BUMPER:
+			{
+				FIXES_gsBumpers[vehicleid][1] = componentid;
+			}
+			case CARMODTYPE_FRONT_BULLBAR:
+			{
+				FIXES_gsBullBars[vehicleid][0] = componentid;
+			}
+			case CARMODTYPE_REAR_BULLBAR:
+			{
+				FIXES_gsBullBars[vehicleid][1] = componentid;
+			}
+		}
+		return AddVehicleComponent__(vehicleid, componentid);
```

Comment:
Call this first and check the return value before assigning variables.

Reply:
Done.